src/runtime/step_runner.js

```diff
@@ -8,22 +8,30 @@ const {beginTiming, endTiming} = Time
 
 async function run({attachmentManager, defaultTimeout, scenarioResult, step, stepDefinition, parameterRegistry, world}) {
   beginTiming()
-  const parameters = stepDefinition.getInvocationParameters({scenarioResult, step, parameterRegistry})
-  const timeoutInMilliseconds = stepDefinition.options.timeout || defaultTimeout
-
-  let error, result
-  const validCodeLengths = stepDefinition.getValidCodeLengths(parameters)
-  if (_.includes(validCodeLengths, stepDefinition.code.length)) {
-    const data = await UserCodeRunner.run({
-      argsArray: parameters,
-      fn: stepDefinition.code,
-      thisArg: world,
-      timeoutInMilliseconds
-    })
-    error = data.error
-    result = data.result
-  } else {
-    error = stepDefinition.getInvalidCodeLengthMessage(parameters)
+  let error, result, parameters
+
+  try {
+    parameters = stepDefinition.getInvocationParameters({scenarioResult, step, parameterRegistry})
+  } catch(err) {
+    error = err
+  }
+
+  if(!error) {
```

Comment:
I do wish `eslint` caught this but regardless I would still like a space added here.